Library/Homebrew/cmd/update-report.rb

```diff
@@ -209,8 +209,12 @@ def report
         end
         @report[:M] << tap.formula_file_to_name(src)
       when /^R\d{0,3}/
-        @report[:D] << tap.formula_file_to_name(src) if tap.formula_file?(src)
-        @report[:A] << tap.formula_file_to_name(dst) if tap.formula_file?(dst)
+        src_full_name = tap.formula_file_to_name(src)
+        dst_full_name = tap.formula_file_to_name(dst)
+        # Don't report formulae that are moved within a tap but not renamed
+        next if src_full_name == dst_full_name
+        @report[:D] << src_full_name if tap.formula_file?(src)
+        @report[:A] << dst_full_name if tap.formula_file?(dst)
```

Comment:
Thanks for finding an even more elegant solution and particularly for adding a nice test case!